chainbase/src/main/java/org/tron/common/storage/leveldb/LevelDbDataSourceImpl.java

```diff
@@ -142,11 +142,22 @@ private void openDatabase(Options dbOptions) throws IOException {
     if (!Files.isSymbolicLink(dbPath.getParent())) {
       Files.createDirectories(dbPath.getParent());
     }
-    database = factory.open(dbPath.toFile(), dbOptions);
-    if (!this.getDBName().startsWith("checkpoint")) {
-      logger.info("DB {} open success with writeBufferSize {} M, cacheSize {} M, maxOpenFiles {}.",
-          this.getDBName(), dbOptions.writeBufferSize() / 1024 / 1024,
-          dbOptions.cacheSize() / 1024 / 1024, dbOptions.maxOpenFiles());
+    try {
+      database = factory.open(dbPath.toFile(), dbOptions);
+      if (!this.getDBName().startsWith("checkpoint")) {
+        logger
+            .info("DB {} open success with writeBufferSize {} M, cacheSize {} M, maxOpenFiles {}.",
+                this.getDBName(), dbOptions.writeBufferSize() / 1024 / 1024,
+                dbOptions.cacheSize() / 1024 / 1024, dbOptions.maxOpenFiles());
+      }
+    } catch (IOException e) {
+      if (e.getMessage().contains("Corruption:")) {
+        throw new RuntimeException(String.format("Can't initialize database: %s, please delete " +

```

Comment:
Please do not throw an exception here. If you do not know what to do, please refer https://github.com/tronprotocol/java-tron/blob/develop/framework/src/main/java/org/tron/core/db/Manager.java#L572. And, please keep the changes to the RocksDB.

Reply:
> Please do not throw an exception here. If you do not know what to do, please refer https://github.com/tronprotocol/java-tron/blob/develop/framework/src/main/java/org/tron/core/db/Manager.java#L572. And, please keep the changes to the RocksDB.

Just exit when open db fail is ok.@halibobo1205